### PR TITLE
Add pgmout and errfile and simplify/clean-up some scripts for EE2 compliance

### PR DESCRIPTION
**Pull request by chan-hoo**
## DESCRIPTION OF CHANGES: 

- Add `pgm`, `pgmout`, `errfile` of `prod_util` to the scripts for the EE2 compliance.
- Remove some arguments from ex-script calls.
- Simplify the scripts for make_grid and make_orog.
- Move mkdir of working directory from ex-scripts to J-job.

## TESTS CONDUCTED: 

- On machines/platforms:

- [x] WCOSS2
- [x] Hera
- [x] Orion
- [ ] Jet

- Test cases: 

- [x] Non-DA engineering test
- [x] DA engineering test
- [ ] Other sample scripts:

## ISSUE: 

- Fixes the issue(s) mentioned in #69 
- Partially fixes the issue mentioned in #71

### CODE REVIEW

**Comment by chan-hoo** [comment]
This is a draft. Not tested yet.

**Comment by chan-hoo** [comment]
@hu5970, in order to meet the NCO standards, I make all executables, except for `ufs_model` and `gsi`, run directly from the `exec` directory like `APRUN {EXECdir}/{pgm}`. Currently, the executables are copied to the working directory first and run in the directory. My sample DA run was complete successfully on Hera. Do you have any concerns about this? @ShunLiu-NOAA, can you test this PR in your parallel run?

**Comment by ShunLiu-NOAA** [comment]
@chan-hoo The changes look good to me. I will coordinate with Ming to start a version on CACTUS to test the changes.

File: jobs/JRRFS_MAKE_SFC_CLIMO
Comment (MatthewPyle-NOAA) [inline jobs/JRRFS_MAKE_SFC_CLIMO:1]: I realize it probably isn't necessary, but for safety should

`mkdir $workdir
`

be

`mkdir -p $workdir`

File: jobs/JRRFS_RUN_ENKF
Comment (MatthewPyle-NOAA) [inline jobs/JRRFS_RUN_ENKF:1]: Is this intended?

`cd -p ${enkfworkdir}`

I wasn't familiar with the "-p" flag on cd, and when tried it got an error:

```
-bash: cd: -p: invalid option
cd: usage: cd [-L|[-P [-e]] [-@]] [dir]
```

**Review by MatthewPyle-NOAA — COMMENTED** [review]
Could the title for this PR be changed?  It does quite a bit more than add pgmout and errfile now.  Also had small questions on the changes within a couple of files.

File: scripts/exrrfs_run_prepstart.sh [hunk 1]
```diff
@@ -389,7 +385,7 @@ else
   if [ -r "${checkfile}" ] ; then
     cp ${bkpath}/${restart_prefix}coupler.res      bk_coupler.res
     cp ${bkpath}/${restart_prefix}fv_core.res.nc   fv_core.res.nc
-    if [ "${IO_LAYOUT_Y}" == "1" ]; then
+    if [ "${IO_LAYOUT_Y}" = "1" ]; then
       for file in ${filelistn}; do
         if [ ${cycle_subtype} == "spinup" ] ; then
```
Comment (hu5970) [inline scripts/exrrfs_run_prepstart.sh:390]: Should be "CYCLE_SUBTYPE" for line 390 and 402

File: jobs/JRRFS_MAKE_SFC_CLIMO
Comment (chan-hoo) [reply jobs/JRRFS_MAKE_SFC_CLIMO:1]: Agreed. Fixed!

File: jobs/JRRFS_RUN_ENKF
Comment (chan-hoo) [reply jobs/JRRFS_RUN_ENKF:1]: Oops. You're correct. "-p" was a typo. Fixed.

File: scripts/exrrfs_run_prepstart.sh (hunk 1 above)
Comment (chan-hoo) [reply scripts/exrrfs_run_prepstart.sh]: Thanks for finding these! Fixed.

**Review by hu5970 — APPROVED** [review]
I texted this PR on Hera. It works well.

**Comment by hu5970** [comment]
@MatthewPyle-NOAA I think this PR is ready to be merged. It changed many files. We can test it again with RRFS_A on develop after merging.

**Review by MatthewPyle-NOAA — APPROVED** [review]
It looks like this one has been evaluated as much as it can without running the full blown real-time system.  So following the request of @hu5970, will get it merged.

**Pull request merged**